### PR TITLE
step towards a single-entry-point js app

### PR DESCRIPTION
**Pull request by stas-vilchik**

### CODE REVIEW

File: server/sonar-web/src/main/js/app/components/nav/app.js [hunk 1]
```diff
@@ -24,13 +24,13 @@ import ReactDOM from 'react-dom';
...
+import { getGlobalNavigation, getComponentNavigation, getSettingsNavigation } from '../../../api/nav';
 
 export default class App {
   start () {
     const options = window.sonarqube;
 
-    require('../../components/workspace/main');
+    require('../../../components/workspace/main');
 
     return new Promise((resolve) => {
```
Comment (SonarTech) [inline server/sonar-web/src/main/js/app/components/nav/app.js]: MINOR Remove parentheses around the parameter of this arrow function. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=javascript%3AS3524)

**Comment by SonarTech** [comment]
SonarQube analysis reported 5 issues
* CRITICAL 2 critical
* MINOR 3 minor

Watch the comments in this conversation to review them.

#### 4 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. CRITICAL [component-nav-menu.js#L98](https://github.com/SonarSource/sonarqube/blob/7c438b3be65628dabd3e402742f60064ec4ae417/server/sonar-web/src/main/js/app/components/nav/component/component-nav-menu.js#L98): Function has a complexity of 12 which is greater than 10 authorized. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=javascript%3AFunctionComplexity)
1. CRITICAL [component-nav-menu.js#L100](https://github.com/SonarSource/sonarqube/blob/7c438b3be65628dabd3e402742f60064ec4ae417/server/sonar-web/src/main/js/app/components/nav/component/component-nav-menu.js#L100): Reduce the number of conditional operators (8) used in the expression (maximum allowed 3). [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=javascript%3AS1067)
1. MINOR [app.js](https://github.com/SonarSource/sonarqube/blob/7c438b3be65628dabd3e402742f60064ec4ae417/server/sonar-web/src/main/js/app/components/nav/app.js): Rename this file to "App". [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=javascript%3AS3317)
1. MINOR [component-nav-menu.js#L112](https://github.com/SonarSource/sonarqube/blob/7c438b3be65628dabd3e402742f60064ec4ae417/server/sonar-web/src/main/js/app/components/nav/component/component-nav-menu.js#L112): Remove curly braces and "return" from this arrow function body. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=javascript%3AS3524)

File: server/sonar-web/src/main/js/app/components/nav/app.js (hunk 1 above)
Comment (trajano) [inline server/sonar-web/src/main/js/app/components/nav/app.js]: Is this enabled in Sonarqube.com or is it only available for local Sonar installs at the moment?

Comment (stas-vilchik) [inline server/sonar-web/src/main/js/app/components/nav/app.js]: @trajano currently it's available only for local installations. See more on http://docs.sonarqube.org/display/PLUG/GitHub+Plugin

Comment (trajano) [inline server/sonar-web/src/main/js/app/components/nav/app.js]: Thanks just wanted to make sure I wasn't doing anything wrong with https://github.com/trajano/trajano/commit/6d43283d225cb97428922975265442c476de22b6

Comment (trajano) [inline server/sonar-web/src/main/js/app/components/nav/app.js]: Hi sorry to take over this but I just wanted to quickly ask this... I just noticed some env vars and I just want to make sure we're not doing anything wrong before I merge the logic for PR support on Travis on https://github.com/apache/maven/pull/97

https://github.com/axel3rd/files-archiver/pull/5#issuecomment-261673966

Comment (stas-vilchik) [inline server/sonar-web/src/main/js/app/components/nav/app.js]: @trajano hi, this is not the best place to ask such questions. It's better to ask in [SonarQube Google Groups](https://groups.google.com/forum/#!forum/sonarqube), the chance to get a good answer is much bigger.

**Pull request merged**